applications/asset_tracker_v2/src/modules/modem_module.c

```diff
@@ -228,6 +244,141 @@ static void modem_rsrp_handler(char rsrp_value)
+			"No error",
+		[LWM2M_CARRIER_ERROR_BOOTSTRAP] =
+			"Bootstrap error",
+		[LWM2M_CARRIER_ERROR_CONNECT_FAIL] =
+			"Failed to connect to the LTE network",
+		[LWM2M_CARRIER_ERROR_DISCONNECT_FAIL] =
+			"Failed to disconnect from the LTE network",
+		[LWM2M_CARRIER_ERROR_FOTA_PKG] =
+			"Package refused from modem",
+		[LWM2M_CARRIER_ERROR_FOTA_PROTO] =
+			"Protocol error",
+		[LWM2M_CARRIER_ERROR_FOTA_CONN] =
+			"Connection to remote server failed",
+		[LWM2M_CARRIER_ERROR_FOTA_CONN_LOST] =
+			"Connection to remote server lost",
+		[LWM2M_CARRIER_ERROR_FOTA_FAIL] =
+			"Modem firmware update failed",
+		[LWM2M_CARRIER_ERROR_CONFIGURATION] =
+			"Illegal object configuration detected",
+	};
+
+	__ASSERT(PART_OF_ARRAY(strerr, &strerr[err->code]), "Unhandled carrier library error");
+
+	LOG_ERR("%s, reason %d\n", strerr[err->code], err->value);
+}
+
+static void print_carrier_deferred_reason(const lwm2m_carrier_event_t *evt)
+{
+	const lwm2m_carrier_event_deferred_t *def = (lwm2m_carrier_event_deferred_t *)evt->data;
+	static const char *const strdef[] = {
+		[LWM2M_CARRIER_DEFERRED_NO_REASON] =
+			"No reason given",
+		[LWM2M_CARRIER_DEFERRED_PDN_ACTIVATE] =
+			"Failed to activate PDN",
+		[LWM2M_CARRIER_DEFERRED_BOOTSTRAP_NO_ROUTE] =
+			"No route to bootstrap server",
+		[LWM2M_CARRIER_DEFERRED_BOOTSTRAP_CONNECT] =
+			"Failed to connect to bootstrap server",
+		[LWM2M_CARRIER_DEFERRED_BOOTSTRAP_SEQUENCE] =
+			"Bootstrap sequence not completed",
+		[LWM2M_CARRIER_DEFERRED_SERVER_NO_ROUTE] =
+			"No route to server",
+		[LWM2M_CARRIER_DEFERRED_SERVER_CONNECT] =
+			"Failed to connect to server",
+		[LWM2M_CARRIER_DEFERRED_SERVER_REGISTRATION] =
+			"Server registration sequence not completed",
+		[LWM2M_CARRIER_DEFERRED_SERVICE_UNAVAILABLE] =
+			"Server in maintenance mode",
+	};
+
+	__ASSERT(PART_OF_ARRAY(strdef, &strdef[def->reason]),
+		"Unhandled deferred carrier library error");
+
+	LOG_ERR("Reason: %s, timeout: %d seconds\n", strdef[def->reason], def->timeout);
+}
+
+int lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *evt)
+{
+	int err = 0;
+
+	switch (evt->type) {
+	case LWM2M_CARRIER_EVENT_MODEM_INIT:
+		LOG_DBG("LWM2M_CARRIER_EVENT_MODEM_INIT");
+		SEND_EVENT(modem, MODEM_EVT_CARRIER_INITIALIZED);
+		break;
+	case LWM2M_CARRIER_EVENT_CONNECTING:
+		LOG_DBG("LWM2M_CARRIER_EVENT_CONNECTING");
+		break;
+	case LWM2M_CARRIER_EVENT_CONNECTED:
+		LOG_DBG("LWM2M_CARRIER_EVENT_CONNECTED");
+		break;
+	case LWM2M_CARRIER_EVENT_DISCONNECTING:
+		LOG_DBG("LWM2M_CARRIER_EVENT_DISCONNECTING");
+		break;
+	case LWM2M_CARRIER_EVENT_DISCONNECTED:
+		LOG_DBG("LWM2M_CARRIER_EVENT_DISCONNECTED");
+		break;
+	case LWM2M_CARRIER_EVENT_BOOTSTRAPPED:
+		LOG_DBG("LWM2M_CARRIER_EVENT_BOOTSTRAPPED");
+		break;
+	case LWM2M_CARRIER_EVENT_LTE_READY: {
+		LOG_DBG("LWM2M_CARRIER_EVENT_LTE_READY");
+		SEND_EVENT(modem, MODEM_EVT_LTE_CONNECTED);
+		break;
+	}
+	case LWM2M_CARRIER_EVENT_REGISTERED:
+		LOG_DBG("LWM2M_CARRIER_EVENT_REGISTERED");
```

Comment:
I would increase this to LOG_INF. Makes it easier to verify at a glance that adding carrier overlay worked as intended. (Successfully registered to  the device management servers).